Tests/src/PublicAPITests/UserSystemTests.cpp

```diff
@@ -233,6 +233,20 @@ CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, ForgotPasswordTest)
 }
 #endif
 
+#if RUN_ALL_UNIT_TESTS || RUN_USERSYSTEM_TESTS || RUN_USERSYSTEM_RESETPASSWORD_TEST
+CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, ResetPasswordTest)
+{
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	auto [Result] = AWAIT_PRE(UserSystem, UserSystem::ResetUserPassword, RequestPredicate, "badtoken", UserId, "NewPassword");
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Failed);
```

Comment:
Why do we expect this to fail?

Reply:
Ah I missed the "badToken". I think the test should be renamed to reflect that you are testing reset password with a bad token.